app/Http/Controllers/API/WpOrg/Themes/ThemeController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Controllers\API\WpOrg\Themes;
+
+use App\Http\Controllers\Controller;
+use App\Models\ApiResultsResponse;
+
+class ThemeController extends Controller
+{
+    private $actions = [
+        'query_themes' => 'doQueryThemes',
+    ];
+
+    public function sendResponse($data, $statusCode = 200)
+    {
+        $version = request()->route('version');
+        if ($version == '1.0') {
+            if (is_object($data) && method_exists($data, 'toStdClass')) {
+                $data = $data->toStdClass();
+            }
+            return response(serialize((object) $data), $statusCode);
+        }
+        return response()->json($data, $statusCode);
+    }
+
+    public function info()
+    {
+        // version is passed as route parameter
+        $action = request()->query('action');
+
+        if (!array_key_exists($action, $this->actions)) {
+            return $this->sendResponse(['error' => 'Action not implemented. <a href="https://codex.wordpress.org/WordPress.org_API">API Docs</a>";}'], 404);
+        }
+        $actionMethod = $this->actions[$action];
+        $requestData = request()->query('request');
+
+        $response = $this->$actionMethod($requestData);
+
+        return $this->sendResponse($response);
+    }
+
+    private function doQueryThemes($requestData)
+    {
+        $page = request()->input('page', 1);
+        ;
+        $perPage = $requestData['per_page'];
+        $skip = ($page - 1) * $perPage;
+        $themes = \DB::table('themes')->skip($skip)->take($perPage)->get()->toArray();
```

Comment:
Please import Facades explicitly, don't use the global aliases which I'll probably actually remove soon.